src/deroff.rs

```diff
@@ -1028,19 +1044,107 @@ impl Deroffer {
         }
     }
 
-    fn do_line(&self, s: &str) -> bool {
-        match s.chars().nth(0) {
-            Some('.') | Some('\'') => !request_or_macro(s),
-            Some(c) => {
+    fn do_tbl(&mut self) -> bool {
+        match self.tblstate {
+            TblState::Options => {
+                while !self.s.is_empty() && !"\n;".contains(self.str_at(0)) {
+                    self.skip_leading_whitespace();
+
+                    if !self.str_at(0).chars().all(|c| c.is_alphabetic()) {
+                        self.skip_char(1);
+                    } else {
+                        let mut option = self.s.to_owned();
+                        let mut arg = String::new();
+
+                        let mut idx: usize = 0;
+
+                        while option
+                            .get(idx..=idx)
+                            .and_then(|s| Some(s.chars().all(|c| c.is_alphabetic())))
+                            .unwrap_or_default()
+                        {
+                            idx += 1;
+                        }
+
+                        if option.get(idx..=idx) == Some("(") {
+                            option = option.get(..idx).unwrap_or_default().to_owned();
+                            self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
+                            arg = self.s.clone();
+                        } else {
+                            self.s = "".to_owned();
+                        }
+
+                        if !arg.is_empty() {
+                            if let Some(idx_) = arg.find("(") {
+                                idx = idx_;
+                                arg = arg.get(..idx).unwrap_or_default().to_owned();
+                            }
+                            self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
```

Comment:
In the python code, `idx` is mutated to `-1` if it is not found, whereas we do not.
```python
idx = arg.find(")")
if idx != -1:
    arg = arg[:idx]
self.s = self.s[idx + 1 :]
```
For the last line, our code may behave different when it is not found.
```suggestion
                            if let Some(idx) = arg.find("(") {
                                arg = arg.get(..idx).unwrap_or_default().to_owned();
                                self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
                            }
```